### ISSUE TITLE
Symbol hyperlinks

### ISSUE THREAD

**Issue by temportalflux** [mersinvald/aquamarine#18]
Mermaid has support for interactions (clicks) on flow-chart nodes (https://mermaid-js.github.io/mermaid/#/flowchart?id=interaction) and rustdocs can generate [links to items by name](https://doc.rust-lang.org/rustdoc/linking-to-items-by-name.html). I'd be very helpful to be able to use both of these concepts and allow crate users to add links to nodes in mermaid syntax utilizing rustdocs item linkage syntax.
Something like:
```mermaid
graph Test
  Item1["[Shown Text](rustdocs::symbol::path)"]
```
This WOULD require specifying `securityLevel:'loose'` in the mermaid config.

**Comment by mersinvald** [comment]
@temportalflux do you think it would be sufficient to set `securityLevel: loose` as a default, under a `hyperlink` feature gate? 

I don't want to make it universal default, partly because it's a breaking change, and also because it is a security issue that I'm not qualified to assess, to be completely honest.

**Comment by temportalflux** [comment]
Yeah I think exposing that option would be fine, though I suspect it would only get integration halfway there. Since mermaid seems to want ful url links, there would need to be some mechanism that converts rustdoc links in a mermaid block into true urls. Perhaps a preprocessor that asks rustdoc to return a hyper link given some rustdoc sybtax that aquamarine detects inside a mermaid-code-block.

**Comment by mersinvald** [comment]
I'll see if I can find some heuristic to do that, but it it's something that requires getting information from `rustdoc`, then we're pretty much out of luck

**Comment by temportalflux** [comment]
It looks like resolving hyperlinks via rustdoc might be viable? I just happened upon this tutorial today which seems to do something similar via some unstable/nightly functionality in rustdoc. https://scrabsha.github.io/abcr-issue-0/#the-basic-idea

**Comment by frehberg** [comment]
Maybe this code snippet might be used to implement some kind of intelligent callback function to redirect to the API doc   https://stackoverflow.com/a/73733645
